### PR TITLE
Security add to project

### PR DESCRIPTION
**Pull request by Bulldog2740**

### CODE REVIEW

File: src/main/java/com/springdata/config/WebConfig.java [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
-}
+import com.springdata.interceptor.SoldierInterceptor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
```
Comment (Taniann) [inline src/main/java/com/springdata/config/WebConfig.java]: add blank line here

File: src/main/java/com/springdata/controller/AppController.java [hunk 2]
```diff
@@ -1,4 +1,56 @@
...
+
+    @GetMapping("/soldiers")
+    public String viewHomePage(Model model) {
+        List<Soldier> listSoldiers = service.getAllSoldiers();
+        model.addAttribute("listSoldiers", listSoldiers);
+        return "index";
+    }
+
+    @GetMapping("/soldiers/new")
+    public String createNewStudentForm(@Valid Model model) {
```
Comment (Taniann) [inline src/main/java/com/springdata/controller/AppController.java]: in mapping soldiers in method name student?
it is confused

File: src/main/java/com/springdata/controller/AppController.java [hunk 3]
```diff
@@ -1,4 +1,56 @@
...
+        Soldier soldier = new Soldier();
+        model.addAttribute("student", soldier);
+        return "new_soldier";
+    }
+
+    @PostMapping("/soldiers/save")
+    public String saveStudent(@ModelAttribute("student") Soldier soldier) {
```
Comment (Taniann) [inline src/main/java/com/springdata/controller/AppController.java]: tha same

File: src/main/java/com/springdata/service/RoleServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class RoleServiceImpl implements RoleService {
+    private final RoleRepo roleRepository;
+
+    @Override
+    public Role findByRole(String role) {
+        return roleRepository.findByRole(role).orElse(new Role());
```
Comment (Taniann) [inline src/main/java/com/springdata/service/RoleServiceImpl.java]: in orElse throw custom exception

File: src/main/java/com/springdata/service/SoldierServiceimpl.java [hunk 5]
```diff
@@ -1,4 +1,36 @@
...
+    }
+
+    @Override
+    public void saveSoldier(Soldier soldier) {
+        repository.save(soldier);
+    }
+
+    @Override
+    public Soldier getSoldierById(Long id) {
+        return repository.findById(id).get();
```
Comment (Taniann) [inline src/main/java/com/springdata/service/SoldierServiceimpl.java]: if optional is empty there will be NPE

File: src/main/java/com/springdata/controller/AppController.java [hunk 6]
```diff
@@ -1,4 +1,56 @@
...
+
+import javax.validation.Valid;
+import java.util.List;
+
+@Controller
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class AppController {
+    private final SoldierService service;
+
+    @GetMapping("/soldiers")
```
Comment (Taniann) [inline src/main/java/com/springdata/controller/AppController.java]: move "soldiers" to the root mapping with annotation RequestMapping

Comment (Bulldog2740) [reply src/main/java/com/springdata/controller/AppController.java]: okey...sorry for so long fixing...today my Birthday))) so i just come home

**Pull request merged**